ecosystem/platform/server/Dockerfile

```diff
@@ -7,48 +7,41 @@ RUN git clone --depth=1 --branch v1.3.0 https://github.com/aptos-labs/aws-env sr
     && cd src/cmd/aws-env \
     && CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -ldflags '-w -s' -o /aws-env && chmod +x /aws-env
 
-FROM ruby:3.1.2-slim
+FROM phusion/passenger-full:2.3.0
 
-RUN apt-get update -qq && apt-get install -yq --no-install-recommends \
-  build-essential \
-  gnupg2 \
-  less \
-  git \
-  libpq-dev \
-  postgresql-client \
-  curl \
-  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+ENV HOME /root
 
-# Install yarn
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg -o /root/yarn-pubkey.gpg && apt-key add /root/yarn-pubkey.gpg
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list
-RUN apt-get update && apt-get install -y --no-install-recommends nodejs yarn
+COPY --link --from=aws-env /aws-env /aws-env
 
-RUN gem update --system && gem install bundler
+RUN npm install -g yarn
 
-WORKDIR /usr/src/app
+# Configure nginx.
+RUN rm -f /etc/service/nginx/down
+RUN rm -f /etc/nginx/sites-enabled/default
 
-COPY --link Gemfile Gemfile.lock ./
+# Copy app to /home/app/webapp.
+RUN mkdir /home/app/webapp
+ADD nginx.conf /etc/nginx/sites-enabled/webapp.conf
+ADD nginx-env.conf /etc/nginx/main.d/webapp-env.conf
+COPY --chown=app:app . /home/app/webapp
+RUN chown app: /home/app/webapp
 
+USER app
+ENV HOME /home/app
 ENV RAILS_ENV production
-ENV RAILS_SERVE_STATIC_FILES true
-ENV RAILS_LOG_TO_STDOUT true
+WORKDIR /home/app/webapp
 
-RUN bundle config --global frozen 1
+RUN bundle config frozen 1
 RUN bundle config set --local without 'development test'
 RUN bundle install
 
-COPY --link . /usr/src/app
-
 # this is a fake secret key just to get it to compile the assets
 ENV SECRET_KEY_BASE=18ff7d41b9da02d394434e2eb140611516707334b7d7a7e15cf8b567061f30e9400a82e61aa772b9e6ccb72853932769d03bbbb9d78f62333c8f7adb95cc727d
 
 # Yes, this should run twice
 # https://github.com/rails/tailwindcss-rails/issues/158
 RUN SKIP_DB_CHECK=1 bundle exec rake assets:precompile && SKIP_DB_CHECK=1 bundle exec rake assets:precompile
 
-COPY --link --from=aws-env /aws-env /aws-env
-
+USER root
+CMD ["/aws-env", "exec", "--", "/sbin/my_init"]
```

Comment:
lets remove anything cloud specific from our containers; worst case we override the CMD when running the image it from a cloud context

Reply:
Should we also remove it from the current image?

https://github.com/aptos-labs/aptos-core/blob/93f81c4a34978daf29e1fef54185ea86be0392a7/ecosystem/platform/server/Dockerfile#L54

Reply:
don't remove it just yet. This is required for the container to be able to load secrets on AWS AppRunner which doesn't support secret injection natively. Once we move to GCP we can remove this.

Reply:
Done.